integrations/spark/spark-3.1/openhouse-spark-runtime/src/main/scala/com/linkedin/openhouse/spark/sql/catalyst/parser/extensions/OpenhouseSqlExtensionsAstBuilder.scala

```diff
@@ -149,7 +150,7 @@ class OpenhouseSqlExtensionsAstBuilder (delegate: ParserInterface) extends Openh
   }
 
   override def visitDuration(ctx: DurationContext): (String, Int) = {
-    val granularity = if (ctx.RETENTION_DAY != null) {
+    val granularity: String = if (ctx.RETENTION_DAY != null) {
```

Comment:
What level of granularity is supported like hour and day only? What if someone provides months, is there a validation not to support that?

Reply:
Maximums is going to be validated at the API level I'm thinking: https://github.com/linkedin/openhouse/pull/259